OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -312,7 +317,7 @@ bool CellIsEvacuating(Actor self, CPos cell)
 
 		public override void Cancel(Actor self, bool keepQueue = false)
 		{
-			if (path != null)
+			if (path != null && mobile.CanStayInCell(mobile.ToCell))
```

Comment:
IMO its a bit weird to have some cases clear the path here and others in `Tick`.
Is there any harm in removing this one completely?

Reply:
See  #16978. I had already removed it once before, but that turned out to be a mistake because of the weird way that `MovePart` interacts with the activity queue. I removed it again downstream after refactoring `MovePart`, but that code is far from mergeable right now.

Reply:
Ok. Can you please add a comment here to explain this?

Reply:
Done.